src/System.Collections/src/System/Collections/BitArray.cs

```diff
@@ -250,9 +241,11 @@ public void Set(int index, bool value)
         public void SetAll(bool value)
         {
             int fillValue = value ? -1 : 0;
-            for (int i = 0; i < m_array.Length; i++)
+            int[] array = m_array;
+
+            for (int i = 0; i < array.Length; i++)
             {
-                m_array[i] = fillValue;
+                array[i] = fillValue;
             }
```

Comment:
`m_array.AsSpan().Fill(fillValue)`? Just like in the BitArray's ctor

Reply:
@ahsonkhan  analyzed it [here](https://github.com/dotnet/corefx/pull/33367#pullrequestreview-174514530)

Reply:
in my pet-project I did `if (length > 10) array.AsSpan().Fill(value); else -> switch=jump table.

Reply:
something like this:
```csharp
switch ((uint)array.Length)
{
    case 8: array[7] = fillValue; goto case 7;
    case 7: array[6] = fillValue; goto case 6;
    case 6: array[5] = fillValue; goto case 5;
    case 5: array[4] = fillValue; goto case 4;
    case 4: array[3] = fillValue; goto case 3;
    case 3: array[2] = fillValue; goto case 2;
    case 2: array[1] = fillValue; goto case 1;
    case 1: array[0] = fillValue; return;
    case 0: return;
}
array.AsSpan().Fill(fillValue);
```

Reply:
The idea is worth a PR :)

Reply:
Duff's device in c#.

Reply:
@Wraith2 from Wikipedia: `When numerous instances of Duff's device were removed from the XFree86 Server in version 4.0, there was an improvement in performance and a noticeable reduction in size of the executable` 😕